codec-http/src/main/java/io/netty/handler/codec/http/HttpHeadersEncoder.java

```diff
@@ -44,23 +45,15 @@ public static void encoderHeader(CharSequence name, CharSequence value, ByteBuf
 
     private static void writeAscii(ByteBuf buf, int offset, CharSequence value, int valueLen) {
         if (value instanceof AsciiString) {
-            writeAsciiString(buf, offset, (AsciiString) value, valueLen);
+            ByteBufUtil.copy((AsciiString) value, 0, buf, offset, valueLen);
         } else {
             writeCharSequence(buf, offset, value, valueLen);
         }
     }
 
-    private static void writeAsciiString(ByteBuf buf, int offset, AsciiString value, int valueLen) {
-        ByteBufUtil.copy(value, 0, buf, offset, valueLen);
-    }
-
     private static void writeCharSequence(ByteBuf buf, int offset, CharSequence value, int valueLen) {
-        for (int i = 0; i < valueLen; i ++) {
+        for (int i = 0; i < valueLen; ++i) {
```

Comment:
FYI, this will only work for US-ASCII.  Is the assumption that this is ok since HTTP/1.x headers only support US-ASCII?